src/Bicep.LangServer/Configuration/BicepConfigChangeHandlingConfigurationManager.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using Bicep.Core.Analyzers.Linter;
+using Bicep.Core.Configuration;
+using Bicep.LanguageServer.Telemetry;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+
+namespace Bicep.LanguageServer.Configuration
+{
+    public class BicepConfigChangeHandlingConfigurationManager : ConfigurationManager, IBicepConfigChangeHandler
```

Comment:
Why the switch from composition to inheritance? It feels worthwhile to keep the *Handler classes as simple as possible.

Reply:
I was able to switch back to composition in https://github.com/Azure/bicep/pull/8301/commits/a0e78721f261c751cbbc598aae4f0297b8a4aee9, but note that it does require making cache management methods on `ConfigurationManager` public. So long as they are omitted from the `IConfigurationManager` interface, I think that's an acceptable tradeoff.